.github/workflows/e2e-tests-ci.yml

```diff
@@ -79,6 +81,45 @@ jobs:
         run: |
           npm run check
 
+  generate-build-variables:
+    runs-on: ubuntu-22.04
+    needs:
+      - update-initial-status
+    outputs:
+      BRANCH: "${{ steps.branch.outputs.BRANCH }}"
+      BUILD_ID_BASE: "${{ steps.build-id-base.outputs.BUILD_ID_BASE }}"
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          ref: ${{ inputs.commit_sha || github.sha }}
+          fetch-depth: 0
+      - name: ci/generate-branch
+        id: branch
+        run: |
+          PR_LIST=$(curl -L -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "${{ github.api_url }}/repos/${{ github.repository }}/commits/${{ inputs.commit_sha || github.sha }}/pulls" 2>/dev/null || echo -n "")
+          if [ -n "$PR_LIST" ]; then
+            # This commit belongs to one or more PRs.
+            # Prefer displaying the PR names in the Automation Dashboard, instead of the branch name
+            PR_NAMES_LIST=$(echo -n "$PR_LIST" | jq -r 'map(.number) | join("-")')
+            BRANCH="server-pr-${PR_NAMES_LIST}"
```

Comment:
I wonder if the PR number can be taken from `github.event.pull_request.number`

Reply:
We're not able to in this workflow. The `github.event.pull_request` data is only available in workflows triggered by a `pull_request` event, but we're triggering it through a [`workflow_dispatch` event](https://github.com/mattermost/mattermost/pull/26093/files#diff-d3a2eba57b09fd1aa50bd75274f7837d31478ba7ad18089516e703f44f6688c1L5), since it's triggered from Argo Events

Reply:
I see, thanks Mario!